### ISSUE TITLE
Code takes as masks dense ground truth labels on warm-up stage

### ISSUE THREAD

**Issue by sevocrear** [lojzezust/SLR#2]
Good afternoon, @lojzezust 
May be, I misled smth but I created the dataset with weak annotations. I ran _prepare_data.py_ script on the data and got the next dir tree:
```
├── all_list.txt
├── all_weak.yaml
├── all.yaml
├── dextr_masks.json
├── images
├── imus
├── masks_weak
├── objects
├── pa_similarity
├── prior_instance_masks
├── test_list.txt
├── train_list.txt
├── train.yaml
├── val_list.txt
├── val.yaml
└── weak_annotations.json
```

But when I run train_slr.sh on this data, I got an error that there are no **masks** directory. So, I returned back to your dataset, opened masks dir and noticed that it has dense GT labels. It seemed strange for me as SLR is the training algorithm which should use only weak annonation (masks_weak). But in your implementation it seems that SLR uses dense GT labels at warm-up stage. Could you please clarify it for me? Why they are used here? In validation?
masks are used in losses:

```
fl = focal_loss(out['out'], labels['segmentation'], target_scale=self.focal_loss_scale)

separation_loss = torch.tensor(0.0)
if self.separation_loss:
    separation_loss = water_obstacle_separation_loss(
        out['aux'], labels['segmentation'], include_sky=self.separation_loss_sky)

pa_loss = torch.tensor(0.0)
if self.pairwise_affinity_loss:
    ignore_mask = None
    if self.object_loss is not None:
        ignore_mask = labels['objects'].max(1).values
    pa_loss = pairwise_affinity_loss(out['out'], labels['segmentation'], labels['pa_similarity'],
                                     tau=self.pairwise_affinity_loss_tau, target_scale=self.focal_loss_scale, ignore_mask=ignore_mask)

separation_loss = self.separation_loss_lambda * separation_loss
pa_loss = self.pairwise_affinity_loss_lambda *
```

For now, I changed in yamls **mask_dir** to **masks_weak** and started learning. You just use GT to calculate metrics on valid step?

**Comment by lojzezust** [comment]
Sorry for the late reply, I am currently on vacation and haven't been checking the emails too often.

The SLR training should **not** use the original GT masks. The `prepare_data.py` script is suposed to create a `all_weak.yaml` data configuration file, that contains the paths to the subdirectories it uses. For the masks it creates a pseudo_labels dir (`masks_weak` by default) where it stores pseudo labels created from the weak annotations. These are the masks that are used during training if everything is setup correctly.

However the validation does use original masks, to validate the method during training. You can disable that step with argument `--validation false` in `train.py`. If you changed the validation masks to `masks_weak` this should work as well, but the validation results will not be correct anyways.
